src/main/java/fact/features/singlePulse/timeSeriesExtraction/SinglePulseExtractor.java

```diff
@@ -0,0 +1,164 @@
+package fact.features.singlePulse.timeSeriesExtraction;
+
+import fact.Utils;
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+
+import java.util.ArrayList;
+
+/**
+ * The single pulse extractor by sebastian mueller acht.
+ * It extracts the arrival times of single photons recorded by the SiPMs.
+ */
+public class SinglePulseExtractor {
+
+    public static class Config {
+        /**
+         * An input configuration to steer the behaviour of a 
+         * SinglePulseExtractor instance.
+         */
+        public int pulseToLookForLength;
+        public int plateauLength;
+        public int negativePulseLength;
+        public double factSinglePeAmplitudeInMv;
+        public int maxIterations;
+
+        public Config() {
+            /**
+             * Default values suitable for FACT's 2GHz standart DRS sampling.
+             */
+            pulseToLookForLength = 20;
+            plateauLength = 7;
+            negativePulseLength = 300;
+            factSinglePeAmplitudeInMv = 10.0;
+            maxIterations = 250;
+        }       
+    }
+
+    public static class Result {
+        /**
+         * A result container to bundle the output of the SinglePulseExtractor.
+         */
+        public int[] pulseArrivalSlices;
+        public double[] timeSeriesAfterExtraction;
+
+        public int numberOfPulses() {
+            return pulseArrivalSlices.length;
+        }
+
+        public double timeSeriesBaseLine() {
+            DescriptiveStatistics statistics = 
+                new DescriptiveStatistics(timeSeriesAfterExtraction);
+            return statistics.getMean();
+        }    
+    }
+
+    public Config config;
+
+    public double[] pulseToLookFor;
```

Comment:
I'm not sure, but it looks to me like some of these variables could be made final. Just to add a bit more rigor to the class. So nobody gets the idea to change these references.